recipe/meta.yaml

```diff
@@ -72,13 +82,37 @@ outputs:
         - ninja
         - make  # [unix]
         - cmake-no-system
-      host:
-        - {{ pin_subpackage('zstd', exact=True) }}
-      run:
-        - {{ pin_subpackage('zstd', exact=True) }}
+      run_constrained:
+        # do not co-install with shared-only lib;
+        # zstd-static contains the shared libs already anyway
+        - zstd <0.0a0
```

Comment:
The issue is that the previous design was completely going against this.

The goal is:

* Try to compile a package that depends on package `a` and `zstd`.
* `a` links to `zstd` dynamically
* The new package links to `zstd` statically.

Reply:
> The issue is that the previous design was completely going against this.

Respectfully, the previous design was a hack, and demonstrably broken (at least from the POV of cmake). I think it's a bad idea for output containing static+shared build to depend on the shared-only build - it all but guarantees broken CMake metadata and/or clobbered files.

> The goal is:
>  * Try to compile a package that depends on package `a` and `zstd`.
>  * `a` links to `zstd` dynamically
>  * The new package links to `zstd` statically.

I don't understand your point here. Why should a package depending on the (already-now) shared `zstd` end up linking statically?

Reply:
On the off chance that you're referring to "zstd-static contains the shared libs already anyway" - this was previously the case already. In fact, the build breaks when passing both 
```
  ZSTD_BUILD_STATIC=ON
  ZSTD_BUILD_SHARED=OFF
```

The error was something like the CLI cannot be built without having the shared lib. I didn't investigate further how to disable building the CLI, as I wanted to keep the previous setup of having both shared & static libs within the `zstd-static` output.

Reply:
> The error was something like "the CLI cannot be built without the shared lib". I didn't investigate further how to disable building the CLI, as I wanted to keep the previous setup of having both shared & static libs within the `zstd-static` output - the only point I changed is that it doesn't depend on the `zstd` output anymore.

I did investigate this now, details in [this](https://github.com/conda-forge/zstd-feedstock/pull/62#issuecomment-1200949913) comment resp. https://github.com/conda-forge/zstd-feedstock/pull/62/commits/22966cd41352f0f96396039f75198fe04d56db56